src/robots/upper_body.cpp

```diff
@@ -239,12 +263,27 @@ class UpperBody: public cardsflow::kindyn::Robot{
         }
 
         vector<float> _integral(motor_ids.size(), 0);
+        vector<float> _error(motor_ids.size(), 0);
         integral[name] = _integral;
+        error_last[name] = _error;
 
+        update();
 
-//        }
 
-        update();
+        if(this->external_robot_state) {
+            // Set current state to bullet
+            publishBulletTarget(name, "current");
+
+            t0 = ros::Time::now();
+            int seconds = 3;
+            while ((ros::Time::now() - t0).toSec() < 3) {
+                ROS_INFO_THROTTLE(1, "waiting %d for setting bullet", seconds--);
+            }
+
+            // Move back to zero position
+            publishBulletTarget(name, "zeroes");
```

Comment:
having a string here is not great. at least an enum or a const string which is not pretty either

Reply:
Resolved in `fix_L_matrix` branch